File: 118. Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []
        for i in range(1, numRows+1):
            res.append(self.makelist(i))
        return res
        
    def fac(self, n):
        res = 1
        for i in range(1,n+1):
            res *= i
        return res
    
    def makelist(self, row):
        if row == 1:
            return [1]
        res = []
        for i in range(row):
            res.append(self.fac(row-1) // self.fac(i) // self.fac(row-i-1))
        return res

File: 118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        if (numRows==0) {
            return ans;
        }
        List<Integer> temp = new ArrayList<>();
        temp.add(1);
        ans.add(new ArrayList(temp));
        if (numRows==1) {
            return ans;
        }
        for (int i=1;i<numRows; i++) {
            List<Integer> last = ans.get(i-1);
            temp = new ArrayList<>();
            temp.add(1);
            for (int j=1; j<i; j++) {
                temp.add(last.get(j-1)+last.get(j));
            }
            temp.add(1);
            ans.add(new ArrayList(temp));
        }
        return ans;
    }
}